make_count/make_count.tex

```diff
@@ -0,0 +1,293 @@
+    
+    Ok(Transformation::new(
+        VectorDomain::new_all(),
+        AllDomain::new(),
+        // think of this as: min(arg.len(), TO::MAX_CONSECUTIVE)
+        Function::new(move |arg: &Vec<TIA>|
+            TO::exact_int_cast(arg.len()).unwrap_or(TO::MAX_CONSECUTIVE)),
+        SymmetricDistance::default(),
+        AbsoluteDistance::default(),
+        StabilityRelation::new_from_constant(TO::one())))
+}
+\end{lstlisting}
+
+\subsection{Implementation of MakeCount in Python-style pseudocode, with preconditions}
+\label{sec:python-pseudocode}
+
+We now use Python-style pseudocode to present a representation of the Rust function.
+
+\emph{Recall that functions in the pseudocode are defined in the document \href{https://www.overleaf.com/project/60d215bf90b337ac02200a99}{``List of definitions used in the pseudocode"}.}
+
+\emph{The use of \texttt{code}-style parameters in the preconditions section below (for example, \texttt{input\_domain}) means that this information should be passed along to the \texttt{Transformation} constructor.}
+
+Here, we use preconditions to check for traits, and to specify the domains and metrics.
+
+\textbf{Preconditions}
+\begin{itemize}
+    \item \textbf{User-specified types:} The \texttt{make\_count} function takes two inputs: a generic input type \texttt{TIA} for the \texttt{Transformation} (meaning that the input vector to \texttt{Transformation} is of type \texttt{Vec(TIA)}), and a generic output type \texttt{TO} for the \texttt{Transformation}.
+    
+    \begin{itemize}
+        \item \texttt{TO} has traits \texttt{One}, \texttt{ExactIntCast(usize)}, and \texttt{DistanceConstant(IntDistance)}
+        
+        \item \texttt{IntDistance} has trait \texttt{InfCast(TO)}
+        
+        \item \question{The final bullet point above is not needed in this proof, but it is needed in the code so a hint can be constructed (otherwise a binary search would be needed to construct the hint). Should this precondition be included here or not?}
+        
+    \end{itemize}
+\end{itemize}
+
+\textbf{Postconditions:} a \texttt{Transformation} must be returned (i.e. if a \texttt{Transformation} cannot be returned successfully, a runtime error should be returned)
+
+\begin{lstlisting}[language = Python, escapechar=|]
+def MakeCount(TIA, TO):
+
+    input_domain = VectorDomain(AllDomain(TIA))
+    output_domain = AllDomain(TO)
+    input_metric = SymmetricDistance()
+    output_metric = AbsoluteDistance(TO)
+    
+    # give the Transformation the following properties
+    max_value = get_max_consecutive_int(TO) 
+    def function(data: Vec<TIA>) -> TO:|\label{line:TO-output}|
+        try:
+            return exact_int_cast(len(data), TO)
+        except FailedCast:
+            return max_value |\label{line:except-return}|
+    def stability_relation(din:u32, dout:TO) -> bool: |\label{line:din-dout-type}|
+        return 1 * inf_cast(din,TO) <= dout |\label{line:stability-relation}|

```

Comment:
it looks like you are indeed using the inf_cast trait here, not just to construct a chaining hint

Reply:
I think we are not. See comment [here](https://github.com/opendp/whitepapers/pull/3#discussion_r701927832).